### DISCUSSION TITLE
Bug in function powu(x,y)

### DISCUSSION THREAD

**Discussion by themastermindnktp** [PaulRBerg/prb-math#54] [General]
In contract `PRBMathSD59x18`, on [line 624](https://github.com/hifi-finance/prb-math/blob/6d849c97831550e521a4bb363bb82eb78bb0415b/packages/prb-math/contracts/PRBMathSD59x18.sol#L624), you have already modified variable `y` but later on [line 639](https://github.com/hifi-finance/prb-math/blob/6d849c97831550e521a4bb363bb82eb78bb0415b/packages/prb-math/contracts/PRBMathSD59x18.sol#L639), you check whether `y` is odd or even. At that point, `y` is always 0, so the result of the function `powu(x, y)` is always positive even if `x` is negative and `y` is odd.

**Comment by PaulRBerg** [comment]
This is worth of becoming an issue. I locked this discussion and created an issue: where we can continue the conversation: https://github.com/hifi-finance/prb-math/issues/55.
